### ISSUE TITLE
Looking for equivalent of `open_novisit(br.click())

### ISSUE THREAD

**Issue by yacoob** [MechanicalSoup/MechanicalSoup#278]
I'm in the process of migrating https://github.com/yacoob/aib2ofx to python 3. Given that mechanize is not python 3 compatible, I was very happy to find MechanicalSoup. I've just bumped into a problem though.

The page I'm automating has a form which results in download of a csv file. With mechanize, I can do this:

```python
br = mechanize.Browser()
br.select_form(predicate=_attrEquals('id', 'historicalTransactionsCommand'))
response = br.open_novisit(br.click())
```

and end up with csv content in `response`. I don't see a clean way of replicating this with MechanicalSoup. There's `download_link`, but there's no link to reference for this function. Before you ask, I can't get to the csv without form submission - all navigation for that bank is done based on forms :( An alternative might be to use normal `submit_selected()`, then ask `StatefulBrowser` to go "back" without reloading the page - but I don't see any public method to do this.

Is there any sensible way of achieving this? If there isn't - can you please add one? :)

Thanks!

**Comment by moy** [comment]
> Given that mechanize is not python 3 compatible

It wasn't, but it is now.

> there's no link to reference for this function

By "reference", do you mean "reference documentation"? If so, https://mechanicalsoup.readthedocs.io/en/stable/mechanicalsoup.html#mechanicalsoup.StatefulBrowser.download_link

>  I can't get to the csv without form submission

That's the problem: `download_link` only works to follow links. You'd need either

* A `download_submission` function, or perhaps `download_selected(update_state=False)` that would behave like `submit_selected`, but that would not change the state of the browser, or

* A `back` function. This has already been requested but never implemented: #127.

The first should be rather straightforward to implement. The second should be reasonably easy also now that we have the `_BrowserState` class. However, no one got time to do this up to now, so the most reliable way to get this implemented is to submit a pull-request.

**Comment by yacoob** [comment]
All right, I gave it a shot:
https://github.com/MechanicalSoup/MechanicalSoup/compare/master...yacoob:download_only
I've tested it for my specific problem and it seems to be working fine. Please tell me if you like the approach/naming, and if you're happy with it I'll add documentation and create a pull request.

**Comment by moy** [comment]
Your branch looks good to me (BTW, you could have submitted a PR right away, it makes it easier for discussion).

Just one nit: parenthesis are not needed after `assert` (although I see that we already have instances of useless parentheses, but I'll fix them). Just `assert foo == bar` is sufficient.

**Comment by yacoob** [comment]
...and the docs are actually okay, as they're autogenerated from the class docs. Nice. Making that PR now. :D

**Comment by moy** [comment]
Fixed by #280.
